src/App.tsx

```diff
@@ -1,54 +1,74 @@
+
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { PersonList } from './components/PersonList';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [debounceQuery, setDebounceQuery] = useState<string>('');
+
+  const applyDebounceQuery = useCallback(debounce(setDebounceQuery, 1000), []);
+
+  const handleInputQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyDebounceQuery(event.target.value);
+  };
+
+  const filteringPersons = useMemo(() => {
+    return peopleFromServer
+      .filter(person => person.name.toLowerCase()
+        .includes(debounceQuery.toLowerCase()));
+  }, [debounceQuery]);
+
+  const handleSelectedPerson = useCallback(
+    (event: React.MouseEvent, person: Person) => {
+      event.preventDefault();
+
+      setQuery('');
+      setDebounceQuery('');
+      setSelectedPerson(person);
+    }, [],
+  );
 
   return (
     <main className="section">
       <h1 className="title">
-        {`${name} (${born} = ${died})`}
+        {selectedPerson === null ? (
+          'No selected person'
+        ) : (
+          `${selectedPerson.name} (${selectedPerson.born} = ${selectedPerson.died})`
+        )}
       </h1>
 
       <div className="dropdown is-active">
         <div className="dropdown-trigger">
           <input
             type="text"
             placeholder="Enter a part of the name"
+            onChange={handleInputQuery}
+            value={query}
             className="input"
           />
         </div>
 
         <div className="dropdown-menu" role="menu">
           <div className="dropdown-content">
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Bernard Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter Antone Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Haverbeke</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-link">Pieter de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Petronella de Decker</p>
-            </div>
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
+            {filteringPersons.length > 0 ? (
+              debounceQuery && (
+                <PersonList
+                  persons={filteringPersons}
```

Comment:
try to do this:
![image](https://github.com/mate-academy/react_autocomplete/assets/86923089/139388ca-7d10-48b8-861b-35c88aff5a3f)

ask in the chat if you need help